internal/eval/deque.go

```diff
@@ -93,8 +95,94 @@ func (q *DequeBasic) LPop() (string, error) {
 	return x, nil
 }
 
+func (q *DequeBasic) insertElementAfterIndex(element string, idx int) {
+	// enc + data + backlen
+	xb := EncodeDeqEntry(element)
+	xbLen := len(xb)
+
+	if cap(q.buf)-len(q.buf) < xbLen {
+		newArr := make([]byte, len(q.buf)+xbLen, (len(q.buf)+xbLen)*2)
+		copy(newArr[xbLen+idx:], q.buf[idx:])
+		copy(newArr[:idx], q.buf[:idx])
+		copy(newArr[idx:idx+xbLen], xb)
+		q.buf = newArr
+	} else {
+		q.buf = q.buf[:xbLen+len(q.buf)]
+		copy(q.buf[xbLen+idx:], q.buf[idx:])
+		copy(q.buf[:idx], q.buf[:idx])
+		copy(q.buf[idx:idx+xbLen], xb)
+	}
+	q.Length++
+}
+
+func (q *DequeBasic) insertBeforeAfterPivot(element, beforeAfter string, pivotIndexStart int, qIterator *DequeBasicIterator) {
+	if pivotIndexStart == 0 && beforeAfter == Before {
+		q.LPush(element)
+		return
+	}
+	if !qIterator.HasNext() && beforeAfter == After {
+		q.RPush(element)
+		return
+	}
+	idx := pivotIndexStart
+	if beforeAfter == After {
+		idx = qIterator.bufIndex
+	}
+	q.insertElementAfterIndex(element, idx)
+}
+
+func (q *DequeBasic) LInsert(pivot, element, beforeAfter string) (int64, error) {
+	// Check if the deque is empty.
+	if q.Length == 0 {
+		return -1, nil
+	}
+	if beforeAfter != Before && beforeAfter != After {
+		return -1, errors.New("invalid before/after argument")
```

Comment:
Redis returns the following:
`(error) syntax error`

Please add a unit test for this edge case.

Reply:
Done.